src/transformers/trainer.py

```diff
@@ -3610,8 +3612,11 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
             labels = inputs.pop("labels")
         else:
             labels = None
-        # if num_items_in_batch is not None:
-        #     inputs["num_items_in_batch"] = num_items_in_batch
+        if self.model_accepts_loss_kwargs:
```

Comment:
This if condition doesn't seem to work for `PeftModel` class (it only has `kwargs` not `loss_kwargs` 🫠 )

I tried just changing that condition to `if True` and ran some tests, and the loss calculation worked perfectly for a LORA on a Llama 3 1B.

I'm wondering if there's a safe/non-breaking way to support peft models here as well?

Reply:
Thanks for the flag, that may be why I couldn't recreate https://github.com/huggingface/transformers/issues/34263#issuecomment-2427711753

Reply:
@man-shar can you try giving peft another go? Should have fixed it

Reply:
Can confirm `self.model_accepts_loss_kwargs` is the correct value now for a peft model!

But seems like the `compute_loss` function inside trainer.py isn't getting the `num_items_in_batch` argument passed to it from `training_step`.

I notice that argument was removed in 4f3f86d2517e766a5773afb0e58c8bf95df10138 and the commit still says "Experimental". So I assume it will be reverted once you're done experimenting! It should work after that!

Reply:
Yes indeed :)